~Finish Collecting Arithmetic Tests

jbrockmendel opened the pull request:
De-duplicate fixtures and put nearly all of them in conftest

jbrockmendel committed:
Move test_arithmetic.py --> tests/arithmetic/test_timedelta64.py

jbrockmendel committed:
Move most remaining arithmetic tests

jbrockmendel committed:
collect fixtures in conftest

jbrockmendel committed:
remove unused imports

gfyoung commented on pandas/tests/arithmetic/conftest.py:
```diff
@@ -30,12 +30,21 @@ def zero(request):
 
 
+@pytest.fixture
+def tdser():
```
Perhaps `td_series` ?  At least an underscore between `td` and `ser` would be good.

jbrockmendel replied:
Good idea, will do.

jbrockmendel committed:
rename tdser-->td_series

jbrockmendel committed:
rename idx-->numeric_idx

jreback commented on pandas/tests/arithmetic/conftest.py:
```diff
@@ -47,6 +56,76 @@ def scalar_td(request):
+@pytest.fixture(params=[np.timedelta64(365, 'D'),
+                        pd.Timedelta(365).to_pytimedelta(),
+                        pd.Timedelta(days=365)] + _common_mismatch)
+def mismatched(request):
```
can you make this more descriptive

jbrockmendel replied:
docstring doesn't get the job done?

jreback replied:
no, prefer this to be a more descriptive name

jreback commented on pandas/tests/arithmetic/conftest.py:
```diff
@@ -47,6 +56,76 @@ def scalar_td(request):
+@pytest.fixture(params=[pd.Timedelta(minutes=30).to_pytimedelta(),
+                        np.timedelta64(30, 's'),
+                        pd.Timedelta(seconds=30)] + _common_mismatch)
+def not_hourly(request):
```
I would add _fixture to these

jreback replied:
can you do this

jreback commented on pandas/tests/arithmetic/test_numeric.py:
```diff
@@ -712,6 +733,51 @@ def check(series, other):
         check(tser, 5)
 
 
+class TestUFuncCompat(object):
```
let's think about putting these in a separate file (test_funcs)? (may want to make numeric a subdir, its already getting long).

jbrockmendel replied:
I'll give that some thought.  Preference is to do quick turnarounds with small PRs here since there are lots of bugs to fix.

jreback reviewed (changes requested):
comments on the fixture names. Also want to ensure that individual test files are completely obvious to what they are testing, so may want to add a level of nesting as things grow longer.

jreback commented:
can you rebase

jbrockmendel committed:
Merge branch 'master' of https://github.com/pandas-dev/pandas into arith_tests8

jbrockmendel commented:
Rebased.  After this the tests are pretty much all consolidated, so the next planned follow-up is to paramatrize/de-duplicate.

Also since#22390 just went in, upcoming PR to fix _most_ of the outstanding xfails

jbrockmendel committed:
Merge branch 'master' of https://github.com/pandas-dev/pandas into arith_tests8

jbrockmendel committed:
Merge branch 'master' of https://github.com/pandas-dev/pandas into arith_tests8

jbrockmendel committed:
Merge branch 'master' of https://github.com/pandas-dev/pandas into arith_tests8

gfyoung commented:
@jbrockmendel : Have you addressed all of the comments here?  I know you were waiting on @jreback to review this again, but at first glance, I can't tell if you have addressed everything he mentioned earlier.

jbrockmendel commented:
> Have you addressed all of the comments here? 

I think so, yes.  But since this is a big enough PR that it is hard to keep track, I made #22559 as a subset of this in the hopes it would be easier in smaller pieces.

gfyoung commented:
> I think so, yes.

@jbrockmendel : What about this one [here](https://github.com/pandas-dev/pandas/pull/22350#discussion_r210548778)?  Just double checking! 🙂 

So should we be reviewing / merging this?  Or should we go after #22559 first?

jbrockmendel commented:
> What about this one here? Just double checking!

The suggestion on the fixture names was to stop fiddling for now and address in the next round.  Since there were multiple comments about fixture names, its easy to lose track when one response is aimed at multiple comments.

> So should we be reviewing / merging this? Or should we go after #22559 first?

Best case would be that everyone is happy here and #22559 is unnecessary.  If this is going to be a slog, then I'd like to get #22559 in as this is a blocker for other work.

gfyoung commented:
ping @jreback - could you review this again?

jreback commented on pandas/tests/arithmetic/conftest.py:
```diff
@@ -30,12 +30,21 @@ def zero(request):
+                        pd.RangeIndex(5)],
                 ids=lambda x: type(x).__name__)
-def idx(request):
+def numeric_index(request):
```
can you docstring

jreback commented on pandas/tests/arithmetic/test_numeric.py:
```diff
@@ -42,6 +33,30 @@ def test_operator_series_comparison_zerorank(self):
+        ts = pd.Timestamp.now()
+        df = pd.DataFrame({'x': range(5)})
+        with pytest.raises(TypeError):
+            df > ts
```
commented in another PR these should be in test_datetime

jreback commented:
can you rebsae

jbrockmendel commented:
Closing in favor of #22645.
